### DISCUSSION TITLE
BuyDisplay ER-TFTM070-5 (RA8875) display does not turn on

### DISCUSSION THREAD

**Discussion by gwiethaus** [lovyan03/LovyanGFX#584] [Device Configurations]
Hello Lovyan03 and great community,

Sorry about my English.

I'm having trouble getting my display to work.

First I will position them on the settings I am using

**Display:**
BuyDisplay ER-TFTM070-5, 7" (800x480), RA8875 microcontroller, Integrated resistive touch screen
Power supply 5V and 3.3V digital signal

**Embarked:**
ESP32 WROOM-32, 30 pins

The communication protocol you are using is SPI.

I tested one of the following codes below, to configure the display, but I was not successful with the LovyanGFX graphics library.

```cpp
//file gfx_conf.h

#define LGFX_USE_V1
#include <LovyanGFX.hpp>

class LGFX : public lgfx::LGFX_Device {
 lgfx::Panel_RA8875 _panel_instance;
 lgfx::Bus_SPI _bus_instance;
 lgfx::Light_PWM _light_instance;

public:
 LGFX(void) {
 {
 auto cfg = _bus_instance.config();
 cfg.spi_host = VSPI_HOST;
 cfg.spi_mode = 0;
 cfg.freq_write = 40000000;
 cfg.freq_read = 16000000;
 cfg.spi_3wire = false;
 cfg.use_lock = true;
 cfg.dma_channel = 1;
 cfg.pin_sclk = 18;
 cfg.pin_mosi = 19;
 cfg.pin_miso = 23;
 cfg.pin_dc = -1; //GPIO27 was tested without success
 _bus_instance.config(cfg);
 _panel_instance.setBus(&_bus_instance);
 }

 {
 auto cfg = _panel_instance.config();
 cfg.pin_cs = 14;
 cfg.pin_rst = 33;
 cfg.pin_busy = 27;
 cfg.memory_width = 800;
 cfg.memory_height = 480;
 cfg.panel_width = 800;
 cfg.panel_height = 480;
 cfg.offset_x = 0;
 cfg.offset_y = 0;
 cfg.offset_rotation = 0;
 cfg.dummy_read_pixel = 8;
 cfg.dummy_read_bits = 1;
 cfg.readable = true;
 cfg.invert = false;
 cfg.rgb_order = false;
 cfg.dlen_16bit = false;
 cfg.bus_shared = true;

 _panel_instance.config(cfg);
 }

 setPanel(&_panel_instance);
 }
};

LGFX tft;
```

I am connecting the BL pin on the display directly to 3.3V (on the ESP32).

The display does not turn on.

ESP32 pins have their main VSPI MOSI, MISO, CLK and CS0 PINS

I tried connecting it in other ways without success.

I don't know if it's the ESP32 with problems, or some incompatibility between my ESP32 model and the library.

I tested the display with another third-party library RA8875.h and Arduino UNO r3, the display worked perfectly.

Could you help me by telling me how I connect my display to the ESP32 and the necessary settings to connect the display? Or otherwise, is there a library incompatibility?

Thank you very much in advance

**Comment by tobozo** [comment]
hi,

please see [this post](https://github.com/lovyan03/LovyanGFX/issues/52#issuecomment-1152802280)

**Reply by gwiethaus** [reply]
Yes. I have also done this suggested implementation and it did not work. The display did not turn on the screen.

**Reply by tobozo** [reply]
let's challenge this configuration:

- cfg.dma_channel = 1 << it may be too early to set this
- cfg.pin_mosi and cfg.pin_miso << it's inverted compared to the suggested example, is it by choice?
- cfg.spi_mode = 0  << can't see this in the example, are you sure it is the correct mode?
- cfg.use_lock = true << do you really need that?
- cfg.bus_shared = true << what other device is sharing the bus?
- cfg.pin_rst <<< try `pinMode( 33, OUTPUT )` in your setup, before calling tft.init()

**Reply by gwiethaus** [reply]
Thank you for your considerations dear "tobozo".
I'll analyze what you pointed out and see where I'm going wrong.

I'm also raising the hypothesis that I burned the display. Why do I think this?

1) Initially, as it didn't work with lovyanGFX on the ESP32 WROOM-32, I decided to test it on the Arduino UNO with another library and it worked. I forgot that the UNO's digital data port is 5V and the display only supports 3.3V. The strangest thing is that I left it running on the Arduino for 15 minutes without any problems. The Display jumper settings were 5V, but this should only be the power supply indicated by the user manual.

2) After that, I decided again to test on the ESP32 and again without success.

3) I then returned to the Arduino UNO. Only this time, the display didn't turn on much to my sadness and disappointment. I think I made a huge mess. I had burned him. I'm still not quite sure about that. I have hope!

I will test your considerations tonight. I will then report the result here.

**Reply by gwiethaus** [reply]
Please can you check my code and display connections with the ESP32. The way I did it is documented in the zipped file

**Comment by gwiethaus** [comment]
Hello, sorry for my English, again.

I tested the suggestions without getting any results. the display remains off. I even posted the suggested code on the forum for testing. Without any effect.

I would like to share the code I made and maybe it can help me find the problem. Remembering that I used Visual Studio Code with platformio.

Thank you once again for the support and help.
[Display ER-TFT0M070-5 LovyanGFX.zip](https://github.com/user-attachments/files/16075141/Display.ER-TFT0M070-5.LovyanGFX.zip)

**Comment by gwiethaus** [comment]
Please can you check my code and display connections with the ESP32. The way I did it is documented in the zipped file

**Reply by tobozo** [reply]
- the config file looks fine and well documented
- the ino file looks strange: launching a task then vTaskDelete(NULL) the setup() function ?
- I can't read the pin labels on the pictures
- I don't have powerpoint so I can't check the ppt file

also I don't understand why red and black wires coming from the display are connected to the +/- on the breadboard, I don't see any power there

however if the display doesn't work anymore on the UNO, it's probably hopeless to try with the ESP

but there's still hope: maybe some pins on the UNO have died and the display is fine

anyway I don't know what else to do

**Reply by gwiethaus** [reply]
Dear tobozo friend,

I want to apologize for forgetting the powerpoint detail. I am sending the picture here of the display configuration pins. It is designed with the colors that I made the display connections with the ESP32.

The supply voltage on the display is 5V according to the jump configuration on the board.

The ESP32 is powered by 3.3V.

I checked all voltage values ​​and they are properly configured (5V - power supply Display, and 3.3V - power supply ESP32).

Part of the code I took from this post: [Issue comment](https://github.com/lovyan03/LovyanGFX/issues/52#issuecomment-1174419941)

If you can test my code on a similar display that would be great. You can make any modifications and suggestions you need in the code. This way I can compare and see what I'm doing wrong.

I am attaching the main code and the drawing of the display pins.

[Display ER-TFT0M070-5 LovyanGFX V2.zip](https://github.com/user-attachments/files/16108898/Display.ER-TFT0M070-5.LovyanGFX.V2.zip)

**Reply by tobozo** [reply]
> The supply voltage on the display is 5V according to the jump configuration on the board.

from the pictures it seems the red and black wires are connected to nowhere

![image](https://github.com/lovyan03/LovyanGFX/assets/1893754/25e31df4-d406-4f82-a08d-bd103663f98c)

even though the esp is powered by regulated 3.3v you can still get 5v from the `VIN` pin on the esp32

I'm 99% sure there's a problem with the wiring, unfortunately I can only help with theory since I don't have such display, and it is limited to my poor knowlegde of buses

please confirm the display still works on the UNO, otherwise it's useless to try with ESP

**Reply by gwiethaus** [reply]
hello friend tobozo,

Yes, it's connected. Unfortunately it is not shown in the image. It is connected on the other side to the 5V power supply via USB. I checked the voltage on these pins that you are indicating. Everything is alright.

> even though the esp is powered by regulated 3.3v you can still get 5v from the VIN pin on the esp32

Yes, you are right. I had forgotten that I can power the display via the VIN pin.

I'll try, once again on the ESP32 and then on the Arduino. Unfortunately after that, I have no hope. I believe the display's digital ports were damaged.

**Comment by gwiethaus** [comment]
I'm getting ready to buy another better display. Unfortunately, in Brazil, with the policy of abusive fees imposed by the "Thief" government, international purchases are greatly harming projects. Think about it, something that costs USD 20.00 turns into USD 40.00. Our Real currency (R$) is almost 6:1 to the dollar. This becomes very prohibitive. This way, it becomes more difficult and expensive to do something cool. As people say: This country is not for amateurs.

Either way, I'm very grateful for your help.

**Comment by TangerineDreamer** [comment]
Hi,gwiethaus,
Are you sure that your alimentation supply can handle your display + esp32 ?
I had similar issues with a big 7" display,feeded by my PC USB and nothing lighten on,so i bought a power-supplied USB HUB
that handle up to more than 2 AMPS and suddenly the display lighten on,my consommation was more that 600 mA!!
If you have a  USB multimeter module,sure you will see more that 500 mA.

**Reply by gwiethaus** [reply]
Hello TangerineDreamer,

Thank you for taking some of your time to help me.

In a way you are right. However, the Display is configured to power the screen at 5V. The power supply to the screen is 5V, 2A.

I used a 2A power supply from my cell phone suitably adapted using a connector with wire output to connect the protoboard. I will make the connection directly without going through the breadboard. I will also take the opportunity to see what current consumption the display is requiring.

![Electrical Features](https://github.com/lovyan03/LovyanGFX/assets/58825334/302ff1d0-e11b-48e4-9c4c-93ff2b97cdf3)

The ESP32 is powered by 3.3V via the USB port directly. The signal on the digital pins obviously cannot exceed the maximum current allowed, otherwise I run the risk of burning the ESP32. Therefore, in this part there is no danger, as the signal sent to the display has a consumption within the limit of the ESP32, as long as the ESP32 has not been damaged.

I will do new tests soon. I also got another identical display from a FabLab for parallel testing to check that I hadn't accidentally damaged my display.

Best regard

**Comment by gwiethaus** [comment]
I decided to simplify the connection of the power source.

![Power Supply_1](https://github.com/lovyan03/LovyanGFX/assets/58825334/51f4f550-afa8-4640-8069-766c743e4479)

![Connector USB from Power Supply_1](https://github.com/lovyan03/LovyanGFX/assets/58825334/446eb7d7-eb00-4831-b935-df06940a66f2)

![Direct Connection_1](https://github.com/lovyan03/LovyanGFX/assets/58825334/dd714cc4-559d-4ab2-b18c-ed06f70d43dd)

**Reply by tobozo** [reply]
:warning: Some USB chargers have an auto shutoff if not enough current is being drawn, I pulled my hair many times at that :wink:

**Reply by gwiethaus** [reply]
So dear friend,

As I explained previously, my charger is 2A. It's power well beyond what the display needs. I also tested with another source and still without success.

I'm already 80% sure that the Display input ports that control it must have been damaged.

If I solve it, I'll post here what really happened. I've tried everything obvious.

**Comment by TangerineDreamer** [comment]
Notherless,it could be safe to test with a stronger supply like a  laboratory one,i'm not convainced with the one you show,many
of them defeat. Is the BL pin always right with 3,3v?
another point: maybe the esp32 5V > 3,3V converter is not strong enough to carry the display.
On the setup(); sometimes i see digitalWrite(BL,HIGH);,maybe a direction to look for.
Anyway,good luck :)

**Reply by gwiethaus** [reply]
However it may be safe to test with a stronger supply like lab, I'm not convinced by what you show, many of them defeat. Is the BL pin always right at 3.3v?
another point: perhaps the esp32 5V > 3.3V converter is not strong enough to charge the display.
In configuration(); sometimes I see digitalWrite(BL,HIGH);, perhaps a direction to look for.
Anyway, good luck :)


So TangerineDreamer, I tested it on Arduino also with another RA8875.h library, simpler, External source to supply 5V and 3.3V of 2A. I also used a bidirectional converter to convert the input and output voltage to the display (avoid damaging it, if it isn't already lol). Unfortunately, they also had no effect. The only way is to see someone on the forum who has the same display as me and can test the code and configuration that I gave. This eliminates any issues with configuration and connection problems.

In theory, I believe that the display is damaged, as I had previously accidentally connected it directly to the Arduino's GPIO digital port (5V), which the maximum display supports is 3.3V. I'm almost 80% certain that the display is damaged.

**Comment by gwiethaus** [comment]
Otherwise, I have 2 displays that I got from a fablab, but I haven't tested one that is completely new yet. The other display already came with the dips soldered onto its board. This one didn't care either. I believe that the person accidentally didn't know this and also connected it to the Arduino directly. All I can do now is definitely test this new one in the box (without making the same mistake). And see if I'm stupid and don't know how to turn it on or if there are really serious problems with the display.

In any case, I'm grateful for the help.

**Comment by gwiethaus** [comment]
My 20% hope came true. The display was not damaged. I used a native library (driver) that was on the Buydisplay website for my model. It first worked on the Arduino Mega 2560. I connected it properly using, in addition to MOSI, MISO, SCLK, CS/SS and RESET, the INT pin.

I connected the ESP32 to my PC's USB, placing its GND in common with the breadboard power supply.

The display was powered by this 5V breadboard source that provides the appropriate current for its operation.

After all this, I rewrote, or rather, added a small code to the library to work on the ESP32.

When turning it on I noticed that there was instability in the display, messing up the image at some point. I didn't know what that was. I took the same assembly to my company's computer and the display images were stable, working well in 2 hours when connected directly. Something on my USB, because the USB on my home computer is 4.0 and 3.1. My job is an old PC with USB 2.0. I believe that some failures may be related to the PC's USB and the ESP32 wroom-32 Devkit 1.

I used a logic alizer from my company and it appears that the data was being sent and received correctly.

I'm going to solve this mystery and then I'm going to move forward with the project, moving on to LovyanGFX and comparing whether it's really the LovyanGFX driver that isn't turning on the display (I hope this time it works with Lovyan).

I would like to thank those who helped me in some way.

**Reply by tobozo** [reply]
the develop branch was updated with this pull request, maybe it'll affect your display too

https://github.com/lovyan03/LovyanGFX/pull/589

**Reply by gwiethaus** [reply]
Grateful. I'll wait for new updates then. The version available in the github directory was the one I had tested.

**Comment by gwiethaus** [comment]
Dear @tobozo,

I found out what was wrong.

In reality it was the writing frequency of the SPI protocol of the display microcontroller.

For the RA8875 it is not 40MHz but 4MHz. This value was found when I used the native library from the Buydisplay manufacturer itself. I analyzed the RA8875.h and RA8875.c libraries and found out what was wrong. I simply adjusted the code to use the lovyanGFX and, like magic, it worked.

```
cfg.freq_write = 4000000;   //4 MHz
cfg.freq_read = 1600000;    //1.6 MHz
```
The value of 40MHz was the original in the initialization settings of the display for the lovyanGFX which I lowered to 4MHz as shown above in the code.

**Reply by tobozo** [reply]
there was another PR for this model, see #606
